tracer/src/Datadog.Trace/RemoteConfigurationManagement/ProductConfigChangedEventArgs.cs

```diff
@@ -7,24 +7,63 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
+using Datadog.Trace.Logging;
+using Datadog.Trace.RemoteConfigurationManagement.Protocol;
 using Datadog.Trace.Vendors.Newtonsoft.Json;
 
 namespace Datadog.Trace.RemoteConfigurationManagement
 {
     internal class ProductConfigChangedEventArgs : EventArgs
     {
-        private readonly IEnumerable<byte[]> _configContents;
+        private readonly IEnumerable<NamedRawFile> _configContents;
 
-        public ProductConfigChangedEventArgs(IEnumerable<byte[]> configContents)
+        private Dictionary<string, ApplyDetails> applyStates = new();
+
+        public ProductConfigChangedEventArgs(IEnumerable<NamedRawFile> configContents)
         {
             _configContents = configContents;
         }
 
-        public IEnumerable<T> GetDeserializedConfigurations<T>()
+        public IEnumerable<NamedTypedFile<T>> GetDeserializedConfigurations<T>()
         {
             return
                 _configContents
-                   .Select(bytes => JsonConvert.DeserializeObject<T>(Encoding.UTF8.GetString(bytes)));
+                   .Select(c => new NamedTypedFile<T>(c.Name, JsonConvert.DeserializeObject<T>(Encoding.UTF8.GetString(c.RawFile))));
```

Comment:
Just FYI, this will conflict with https://github.com/DataDog/dd-trace-dotnet/pull/3169

Reply:
Merged and resolved.